packages/web/src/sideEffects/localFs/utils.js

```diff
@@ -1,18 +1,50 @@
+/*
+ * Compare the given files for equality; file path and file contents
+ * @return {Boolean} true if the files are the same
+ */
+const sameFile = (a, b) => {
+  if (a.fullPath === b.fullPath) {
+    if (typeof a.source === typeof b.source) {
+      if (a.source.byteLength) {
+        // compare ArrayBuffer contents
+        const aArray = new Uint8Array(a.source)
+        const bArray = new Uint8Array(b.source)
+        return aArray.length === bArray.length && aArray.every((v, i) => v === bArray[i])
+      } else {
+        // compare String contents
+        return a.length === b.length && a.source === b.source
+      }
+    }
+  }
+  return false
+}
+
 /**
- * returns list of files whose source has changed compared to the file with the same
- * name in memFs
- * @param {Object} reference - The reference hash of files, organized by name.
- * @param {Array} files - The list of files : need to have a source & name field.
- * @return array of files that changed
+ * Compare the contents of the reference and the new file lists.
+ * @see flattenFiles() below
+ * @param {Array} reference - the reference list of files
```

Comment:
Feel like the variable names here could make this more readable:
changedFiles => areFilesChanged or areFilesDifferent
reference => referenceFileList or fileList1
files => newFileList or fileList2

Reply:
Actually, changedFiles is correct, as a list of changed files is returned.

I’ll review the other names...

Reply:
ah, gotcha.  👍